modules/setting/setting.go

```diff
@@ -546,9 +546,27 @@ func SetCustomPathAndConf(providedCustom, providedConf, providedWorkPath string)
 	}
 }
 
-// NewContext initializes configuration context.
+// NewContextFromExistingConf initializes configuration context from an exisint config file (app.ini)
+func NewContextFromExistingConf() {
+	newContextFromConf(false)
+}
+
+// NewContextAllowEmptyConf initializes configuration context, it's also fine that if the config file (app.ini) doesn't exist
+func NewContextAllowEmptyConf() {
+	newContextFromConf(true)
+}
+
+// NewContextForTest initializes configuration context for tests
+func NewContextForTest() {
```

Comment:
These are all a bit wordy but I'm not going to fight.

I think we could get away with:
```suggestion
// NewContext initializes configuration context from an existing config file (app.ini)
func NewContext() {
	newContextFromConf(false)
}

// NewContextAllowEmpty initializes configuration context, but doesn't require a pre-existing configuration file 
func NewContextAllowEmpty() {
	newContextFromConf(true)
}

// NewTestContext initializes configuration context for tests
func NewTestContext() {
```

Reply:
Well, the `Context` is not good here. If we want to rename them, I will propose a totally new name.

Reply:
I renamed them to `setting.LoadFromExisting`, `setting.LoadAllowEmpty` and `setting.LoadForTest`

How do you think? 😊

Reply:
yeah that's better.